packages/pcd/pod-ticket-pcd/src/PODTicketPCDPackage.ts

```diff
@@ -0,0 +1,244 @@
+      type: "string",
+      value: ticket.eventName
+    },
+    ticketName: {
+      type: "string",
+      value: ticket.ticketName
+    },
+    isConsumed: {
+      type: "int",
+      value: ticket.isConsumed ? 1n : 0n
+    },
+    isRevoked: {
+      type: "int",
+      value: ticket.isRevoked ? 1n : 0n
+    },
+    checkerEmail: {
+      type: "string",
+      value: ticket.checkerEmail ?? ""
+    },
+    timestampConsumed: {
+      type: "int",
+      value: BigInt(ticket.timestampConsumed)
+    },
+    timestampSigned: {
+      type: "int",
+      value: BigInt(ticket.timestampSigned)
+    },
+    imageUrl: {
+      type: "string",
+      value: ticket.imageUrl ?? ""
+    },
+    imageAltText: {
+      type: "string",
+      value: ticket.imageAltText ?? ""
+    },
+    ticketCategory: {
+      type: "int",
+      value: BigInt(ticket.ticketCategory)
+    }
+  };
+}
+
+/**
+ * Creates a new {@link PODTicketPCD} by generating an {@link PODTicketPCDProof}
+ * and deriving an {@link PODTicketPCDClaim} from the given {@link PODTicketPCDArgs}.
+ */
+export async function prove(args: PODTicketPCDArgs): Promise<PODTicketPCD> {
+  if (!args.privateKey.value) {
+    throw new Error("missing private key");
+  }
+
+  if (!args.ticket.value) {
+    throw new Error("missing ticket value");
+  }
+
+  const ticket = args.ticket.value;
+
+  const podPCD = await PODPCDPackage.prove({
+    privateKey: args.privateKey,
+    id: {
+      value: undefined,
+      argumentType: ArgumentTypeName.String
+    },
+    entries: {
+      value: ticketDataToEntries(ticket),
+      argumentType: ArgumentTypeName.Object
+    }
+  });
+
+  const id = args.id.value ?? uuid();
+
+  return new PODTicketPCD(id, { ticket }, { podPCD });
+}
+
+/**
+ * Verifies a POD Ticket PCD by checking that its {@link PODTicketPCDClaim} corresponds to
+ * its {@link PODTicketPCDProof}. If they match, the function returns true, otherwise false.
+ * In most cases, verifying the validity of the PCD with this function is not enough.
+ * It may also be necessary to ensure that the parameters of the ticket, such as the
+ * productId and eventId, match the expected values, and that the public key of the
+ * entity that signed the ticket is indeed the authority for that event.
+ */
+export async function verify(pcd: PODTicketPCD): Promise<boolean> {
+  const content = PODContent.fromEntries(ticketDataToEntries(pcd.claim.ticket));
+
+  if (content.contentID !== pcd.proof.podPCD.pod.contentID) {
+    return false;
+  }
+
+  return PODPCDPackage.verify(pcd.proof.podPCD);
+}
+
+/**
+ * Serializes a {@link PODTicketPCD}.
+ * @param pcd The POD Ticket PCD to be serialized.
+ * @returns The serialized version of the POD Ticket PCD.
+ */
+export async function serialize(
+  pcd: PODTicketPCD
+): Promise<SerializedPCD<PODTicketPCD>> {
+  const serializedPODPCD = await PODPCDPackage.serialize(pcd.proof.podPCD);
+
+  return {
+    type: PODTicketPCDTypeName,
+    pcd: JSONBig().stringify({
```

Comment:
Does this implementation actually need JSONBig?
I think IPODTicketData is all natively JSONable, and serializedPODPCD is already a string.